File: src/main/java/niklasu/speedtester/ui/ConsoleResultPrinter.java
package niklasu.speedtester.ui;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ConsoleResultPrinter {
    private final Logger logger = LoggerFactory.getLogger(ConsoleResultPrinter.class);

    public void show(double resultSpeed) {
        logger.info(String.format("%.2f MBit/s", resultSpeed));
    }

    public void showProgress(long current, long total) {
        int amount = (int) (((double) current / (double) total) * 100);
        char[] animationChars = { '|', '/', '-', '\\' };
        System.out.print("Processing: " + amount + "% " + animationChars[amount % 4] + "\r");
    }
}

File: src/main/java/niklasu/speedtester/ui/ConsoleResultPrinter.kt
package niklasu.speedtester.ui

import niklasu.speedtester.logger

class ConsoleResultPrinter {
    private val logger = logger()
    fun show(resultSpeed: Double) {
        logger.info(String.format("%.2f MBit/s", resultSpeed))
    }

    fun showProgress(current: Long, total: Long) {
        val amount = ((current.toDouble() / (total.toDouble())) * 100).toInt()
        val animationChars = charArrayOf('|', '/', '-', '\\')
        System.out.print("Processing: " + amount + "% " + animationChars[amount % 4] + "\r")
    }
}